onnxruntime/core/framework/prepacked_weights_container.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/prepacked_weights_container.h"
+#include "core/framework/allocatormgr.h"
+
+namespace onnxruntime {
+
+class BufferHasher {
+ public:
+  std::size_t operator()(const int8_t* data, size_t num_bytes) const {
+    std::size_t ret = num_bytes;
+    for (size_t i = 0; i < num_bytes; ++i) {
+      ret ^= std::hash<int32_t>()(data[i]);
+    }
+    return ret;
+  }
+};
+
+uint64_t PrepackedWeight::GetHash() {
+  // Adaptation of the hashing logic of the KernelDef class
+
+  uint32_t hash[4] = {0, 0, 0, 0};
+
+  auto hash_size_t = [&hash](size_t i) { MurmurHash3::x86_128(&i, sizeof(size_t), hash[0], &hash); };
+
+  ORT_ENFORCE(buffers_.size() == buffer_sizes_.size());
+  for (size_t iter = 0; iter < buffers_.size(); ++iter) {
+    BufferHasher buffer_hasher;
+    size_t buffer_hash = buffer_hasher(reinterpret_cast<int8_t*>(buffers_[iter].get()), buffer_sizes_[iter]);
+    hash_size_t(buffer_hash);
```

Comment:
Not quite understanding why the extra std::hash layer in BufferHasher is necessary (and it seems flawed as it doesn't do any shifting like in boost::hash_combine).

You're just hashing all the data in the buffers right? So why not just push it all through MurmurHash3 in a big loop and use the output from that?

Reply:
Cleaned up hashing logic